clang/lib/CIR/Lowering/ThroughMLIR/LowerCIRToMLIR.cpp

```diff
@@ -909,6 +927,62 @@ class CIRIfOpLowering : public mlir::OpConversionPattern<mlir::cir::IfOp> {
   }
 };
 
+class CIRSwitchOpLowering
+    : public mlir::OpConversionPattern<mlir::cir::SwitchOp> {
+public:
+  using mlir::OpConversionPattern<mlir::cir::SwitchOp>::OpConversionPattern;
+
+  mlir::LogicalResult
+  matchAndRewrite(mlir::cir::SwitchOp op, OpAdaptor adaptor,
```

Comment:
BTW, I am confused why we need to lower `SwitchOP`? I think, we should deal with `SwitchFlatOp` only since all the `SwitchOP` should be faltten.

Reply:
Thank you for taking the time to review my code. Considering potential polyhedral optimizations, I kept the path using the -emit-cir option. The CIR generated this way provides more information and can be translated into higher-level MLIR dialects, and this option produces the SwitchOp.